salt/modules/aptpkg.py

```diff
@@ -2360,7 +2360,13 @@ def add_repo_key(
     cmd = ["apt-key"]
     kwargs = {}
 
-    current_repo_keys = get_repo_keys(aptkey=aptkey, keydir=keydir)
+    # If the keyid is provided or determined, check it against the existing
```

Comment:
I couldn't see a reason why this code was at the bottom of this function, and so I moved it up to avoid the extra work of caching and decrypting keys if they already exist on disk. However, I could totally be missing something.